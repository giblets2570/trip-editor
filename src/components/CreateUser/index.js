import React, { Component } from 'react'
import { connect } from 'react-redux'
import { 
  Modal, ModalHeader, ModalBody,
  Button, Form,
  FormGroup, Input, Label
} from 'reactstrap'

import { passwordWrong, create, update, remove } from '../../actions/authActions'

class CreateUser extends Component {
  constructor(props){
    super(props)
    this.state = {
      name: "",
      email: "",
      password: "",
      passwordCheck: ""
    }
    this.submit = this.submit.bind(this)
    this.create = this.create.bind(this)
    this.update = this.update.bind(this)
    this.remove = this.remove.bind(this)
  }
  componentWillReceiveProps(nextProps) {
    this.setState({
      name: nextProps.user ? nextProps.user.name : "",
      email: nextProps.user ? nextProps.user.email : "",
      password: nextProps.user && nextProps.user.password ? nextProps.user.password : "",
      passwordCheck: nextProps.user && nextProps.user.passwordCheck ? nextProps.user.passwordCheck : ""
    })
  }
  handleChange(event,key) {
    this.setState({
      [key]: event.target.value
    })
  }
  submit(e) {
    e.preventDefault()
    if(this.state.password !== this.state.passwordCheck){
      this.props.dispatch(passwordWrong())
    }else{
      if(this.props.user && this.props.user._id){
        this.update()
      }else{
        this.create()
      }
    }
    this.props.toggle()
  }
  create(){
    this.props.dispatch(create(this.state))
  }
  update(){
    this.props.dispatch(update(this.props.user._id,this.state))
  }
  remove(){
    this.props.dispatch(remove(this.props.user._id))
    this.props.toggle()
  }
  render(){
    const header = this.props.user ? "Editing user" : "Create a user"
    let deleteButton = null
    if(this.props.user) {
      deleteButton = (
        <Button color="danger" onClick={this.remove}>
          Delete
        </Button>
      )
    }
    return (
      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader 
          color="danger"
          className="green-color"
          toggle={this.props.toggle}>{header}</ModalHeader>
        <ModalBody>
          <Form onSubmit={this.submit}>
            <FormGroup>
              <Label 
                for="Name">Name</Label>
              <Input
                type="text"
                value={this.state.name}
                name="name"
                id="Name"
                onChange={(e) => this.handleChange(e, 'name')}
                placeholder="" />
            </FormGroup>
            <FormGroup>
              <Label 
                for="Email">Email</Label>
              <Input
                type="email"
                value={this.state.email}
                name="email"
                id="Email"
                onChange={(e) => this.handleChange(e, 'email')}
                placeholder="" />
            </FormGroup>
            <FormGroup>
              <Label
                for="Password">New Password</Label>
              <Input
                type="password"
                value={this.state.password}
                name="password"
                id="Password"
                onChange={(e) => this.handleChange(e, 'password')}
                placeholder="" />
            </FormGroup>
            <FormGroup>
              <Label 
                for="PasswordCheck">New Password Check</Label>
              <Input
                type="password"
                value={this.state.passwordCheck}
                name="name"
                id="PasswordCheck"
                onChange={(e) => this.handleChange(e, 'passwordCheck')}
                placeholder="" />
            </FormGroup>
            <Button color="success">
              Submit
            </Button>
            {deleteButton}
          </Form>
        </ModalBody>
      </Modal>
    )
  }
}

export default connect()(CreateUser)