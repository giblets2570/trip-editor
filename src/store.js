import { compose, applyMiddleware, createStore } from "redux"
import { persistStore, autoRehydrate } from "redux-persist"

import { createLogger }  from "redux-logger"
import thunk from "redux-thunk"
import promise from "redux-promise-middleware"

import reducer from "./reducers"

const middleware = applyMiddleware(promise(), thunk, createLogger())

const store = createStore(
	reducer, 
	undefined,
	compose(
		middleware,
		// autoRehydrate()
	)
)

// persistStore(store);

export default store;