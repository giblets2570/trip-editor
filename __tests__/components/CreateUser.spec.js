import React from 'react';
import renderer from 'react-test-renderer';
import CreateUser from '../../src/components/CreateUser';
import store from '../../src/store'
import { Provider } from 'react-redux'

describe('CreateUser (Snapshot)', () => {
  it('CreateUser renders', () => {
    let wrapper = shallow(
      <CreateUser store={mockStore}/>
    );
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.prop('user')).toEqual(undefined);

    const user = {
        name: "Tom",
        role: "user",
        email: "tom@tom.com"
    }
    
    wrapper = shallow(
      <CreateUser store={mockStore} user={user}/>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.prop('user')).toEqual(user);


  });
});