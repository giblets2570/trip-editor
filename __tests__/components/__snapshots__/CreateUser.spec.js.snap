// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser (Snapshot) CreateUser renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CreateUser
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  "nodes": Array [
    <CreateUser
      dispatch={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(CreateUser)
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "dispatch": [Function],
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 0,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CreateUser
          dispatch={[Function]}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        />,
        "_debugID": 2,
        "_renderedOutput": <CreateUser
          dispatch={[Function]}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(CreateUser)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
}
`;

exports[`CreateUser (Snapshot) CreateUser renders 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <CreateUser
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    user={
      Object {
        "email": "tom@tom.com",
        "name": "Tom",
        "role": "user",
      }
    }
  />,
  "nodes": Array [
    <CreateUser
      dispatch={[Function]}
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
      user={
        Object {
          "email": "tom@tom.com",
          "name": "Tom",
          "role": "user",
        }
      }
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(CreateUser)
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        user={
          Object {
            "email": "tom@tom.com",
            "name": "Tom",
            "role": "user",
          }
        }
      />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "user": Object {
            "email": "tom@tom.com",
            "name": "Tom",
            "role": "user",
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "dispatch": [Function],
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "user": Object {
              "email": "tom@tom.com",
              "name": "Tom",
              "role": "user",
            },
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 0,
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <CreateUser
          dispatch={[Function]}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          user={
            Object {
              "email": "tom@tom.com",
              "name": "Tom",
              "role": "user",
            }
          }
        />,
        "_debugID": 4,
        "_renderedOutput": <CreateUser
          dispatch={[Function]}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
          user={
            Object {
              "email": "tom@tom.com",
              "name": "Tom",
              "role": "user",
            }
          }
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(CreateUser)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    user={
      Object {
        "email": "tom@tom.com",
        "name": "Tom",
        "role": "user",
      }
    }
  />,
}
`;
